pkgs/pkgs-lib/formats.nix

```diff
@@ -92,6 +92,8 @@ rec {
     # Alternative to listsAsDuplicateKeys, converts list to non-list
     # listToValue :: [IniAtom] -> IniAtom
     listToValue ? null,
+    # Whether to use toINI or toINIWithGlobalSection
+    withGlobalSection ? false,
```

Comment:
I am a bit worried that this makes it easy to break modules. If they start out with `ini {}`, the change to `ini { withGlobalSection = true; }` would break all assignments.

And we also can't remove the default here because that would break all uses of `ini {}`...